Event Hub pod identity blob storage support (#905)
kedacore/keda-docs: content/docs/2.9/scalers/azure-event-hub.md

    cloud: Private
    # Required when cloud = Private
    endpointSuffix: servicebus.airgap.example
    # Required when cloud = Private
    storageEndpointSuffix: airgap.example
    # Required when cloud = Private.
    activeDirectoryEndpoint: https://login.airgap.example/
    # Required when cloud = Private.
    eventHubResourceURL: https://eventhubs.airgap.example/
    # Required when using pod identity authentication with blob storage
    storageAccountName: 'name_of_account'
```

**Parameter list:**

- `connectionFromEnv` - Name of the environment variable your deployment uses to get the connection string appended with `EntityPath=<event_hub_name>`.

> - `storageConnectionFromEnv` - Name of the environment variable that provides connection string for Azure Storage Account to store checkpoint. As of now the Event Hub scaler only reads from Azure Blob Storage. Do not specify when using with pod identity.

Reviewer: ```suggestion
- `storageConnectionFromEnv` - Name of the environment variable that provides connection string for Azure Storage Account to store checkpoint. As of now the Event Hub scaler only reads from Azure Blob Storage. (Only required when not using pod identity)
```

- `consumerGroup` - Consumer group of Azure Event Hub consumer. (default: `$default`, Optional)
- `unprocessedEventThreshold` - Average target value to trigger scaling actions. (Default: `64`, Optional)
- `activationUnprocessedEventThreshold` - Target value for activating the scaler. Learn more about activation [here](./../concepts/scaling-deployments.md#activating-and-scaling-thresholds).(Default: `0`, Optional)
- `blobContainer` - Container name to store checkpoint. This is needed for every `checkpointStrategy` except of `AzureFunction`. With Azure Functions the `blobContainer` is autogenerated and cannot be overridden.

> - `storageAccountName` - Account name for blob storage used for checkpoints. The storage account name is the first part in the hostname of a Blob Storage endpoint. E.g.: for `examplename.blob.core.windows.net` the account name is `examplename`. The endpoint is automatically configured based on the `cloud` environment set. (Required when using pod identity authentication in the absence of a connection string for `storageConnectionFromEnv`)

Reviewer: Can you align with https://github.com/kedacore/keda-docs#writing-documentation-for-a-scaler please?

Author: Does this fit the guidance?

Reviewer: Yes

    provider: Azure
```

> When you do so, the Event Hub scaler will depend on the existence of two configurations you have to provide: `eventHubNamespace` and `eventHubName` as well as `storageAccountName` if you wish for the Blob Storage client to use the Pod Identity instead of a connection string. 

Reviewer: ```suggestion
When you do so, the Event Hub scaler will depend on the existence of two configurations you have to provide: `eventHubNamespace` and `eventHubName`. You can also configure `storageAccountName` if you wish to use Azure AD Pod Identity to authenticate to Azure Blob Storage instead of a connection string. 
```

    provider: Azure | azure-workload
```

> When you do so, the Event Hub scaler will depend on the existence of two configurations you have to provide: `eventHubNamespace` and `eventHubName`. You can also configure `storageAccountName` if you wish to use Azure AD Pod Identity to authenticate to Azure Blob Storage instead of a connection string. 

Reviewer 2: Sorry for the trouble here, a minor change.

```suggestion
When you do so, the Event Hub scaler will depend on the existence of two configurations you have to provide: `eventHubNamespace` and `eventHubName`. You can also configure `storageAccountName` if you wish to use Azure AD Pod / Workload Identity to authenticate to Azure Blob Storage instead of a connection string. 
```
